Fix update_axes to respect selector duration instead of hardcoded 1000ms

ANAMASGARD opened the pull request:
Fixes #276

## Problem

The `update_axes()` function in the JavaScript renderer was using a hardcoded transition duration of 1000ms, regardless of what duration was set for the selector triggering the axis update.

This meant that when a user specified `duration=list(year=2000)` for a year selector, clicking to change years would cause:
- The data points to transition over 2000ms (correct)
- The axes to transition over 1000ms (incorrect - looks janky)

## Testing

Added `test-renderer3-update-axes-duration.R` which:
- Creates a visualization with `duration=list(year=2000)` 
- Verifies that axis transitions take 2000ms by checking position at 1200ms
- With the bug: transition completes in 1000ms, test fails 
- With the fix: transition takes 2000ms, test passes 

This failing test committed first, then the fix in a upcoming commits .

tdhock reviewed (changes requested):
please fix test failures before asking for review

tdhock commented on inst/htmljs/animint.js:
```diff
@@ -2012,7 +2012,9 @@ var animint = function (to_select, json_file) {
-    if(Selectors.hasOwnProperty(v_name) && Selectors[v_name].duration){
+    if(Selectors.hasOwnProperty(v_name) && 
+       Selectors[v_name].hasOwnProperty("duration") &&
+       typeof Selectors[v_name].duration === "number"){
```
this check is un-necessary because duration is always a number

ANAMASGARD commented:
Sir @tdhock  Can you please review this PR now and give your feedback is there anything I need to change .

tdhock commented on tests/testthat/test-renderer3-update-axes-duration.R:
```diff
@@ -31,8 +31,7 @@ updated_tick_diff_y <- getTickDiff(y_axis_node_updated, axis="y")
-  expect_false(identical(updated_tick_diff_y, original_tick_diff_y),
-               info="Y-axis ticks should change when selector changes")
+  # Use numeric comparison - tick differences should change when selector changes
+  expect_gt(abs(updated_tick_diff_x[2] - original_tick_diff_x[2]), 0.01)
```
why 2 and 0.01? please avoid magic numbers

tdhock commented on tests/testthat/test-renderer3-update-axes-duration.R:
```diff
@@ -31,7 +31,9 @@ updated_tick_diff_y <- getTickDiff(y_axis_node_updated, axis="y")
-  expect_gt(abs(updated_tick_diff_x - original_tick_diff_x), 0.01)
-  expect_gt(abs(updated_tick_diff_y - original_tick_diff_y), 0.01)
+  # Using same pattern as test-renderer3-update-axes.R
+  expect_true(unequal(updated_tick_diff_x, original_tick_diff_x, tolerance=0.01),
```
this is frustrating for me.
I already told you to avoid expect true and false, and now you put it back? 
please remove 0.01 or explain why it is relevant and necessary.

ANAMASGARD commented:
I sincerely apologize Sir you are absolutely right I should have followed your previous feedback 
THE THINGS I HAVE CHANGED THIS TIME 

- Replaced expect_true with expect_gt - a more specific assertion as you requested
 - Removed the 0.01 tolerance - we now simply check that the difference is greater than 0
 - Added a comment explaining why tick spacing changes (different cyl subsets have different mpg/disp ranges)
 - The test now directly asserts that the tick spacing changed (difference > 0) without any magic numbers or expect_true/expect_false.
